client/packages.go

```diff
@@ -76,10 +81,94 @@ func (client *Client) Packages() (map[string]Package, error) {
 		return nil, fmt.Errorf("%s: response has no packages", errPrefix)
 	}
 
-	var packages map[string]Package
-	if err := json.Unmarshal(packagesJSON, &packages); err != nil {
+	var packageMap map[string]Package
+	if err := json.Unmarshal(packagesJSON, &packageMap); err != nil {
 		return nil, fmt.Errorf("%s: failed to unmarshal packages: %v", errPrefix, err)
 	}
 
+	var packages Packages
+	for _, pkg := range packageMap {
+		packages = append(packages, pkg)
+	}
+
 	return packages, nil
 }
+
+// IsInstalled returns true if the Package is currently installed
+func (p Package) IsInstalled() bool {
+	return p.Status == StatusInstalled || p.Status == StatusActive
+}
+
+// HasNameContaining returns true if the Package name contains the query. The
+// comparison is case-insensitive
+func (p Package) HasNameContaining(query string) bool {
+	return strings.Contains(strings.ToLower(p.Name), strings.ToLower(query))
+}
+
+// HasTypeInSet returns true if the Package type is a member of the given list
+// of types
+func (p Package) HasTypeInSet(types []string) bool {
+	for _, t := range types {
+		if p.Type == t {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Installed returns the installed items from Packages
+func (p Packages) Installed() Packages {
```

Comment:
Thanks. I had memories about it supporting sources limiting, but I'm probably thinking of the API docs instead.